docs/content/docs/ops/batch/recovery_from_job_master_failure.md

```diff
@@ -0,0 +1,110 @@
+supports this feature. And Flink batch jobs will use this scheduler by default unless another scheduler is explicitly configured.
+
+### Optimization
+
+To enable batch jobs to recover as much progress as possible after a JobMaster failover, and avoid rerunning tasks 
+that have already been finished, you can configure the following options for optimization:
+
+- [execution.batch.job-recovery.snapshot.min-pause]({{< ref "docs/deployment/config" >}}#execution-batch-job-recovery-snapshot-min-pause):
+  This setting determines the minimum pause time allowed between snapshots for the OperatorCoordinator and ShuffleMaster.
+  This parameter could be adjusted based on the expected I/O load of your cluster and the tolerable amount of state regression. 
+  Reduce this interval if smaller state regressions are preferred and a higher I/O load is acceptable.
+- [execution.batch.job-recovery.previous-worker.recovery.timeout]({{< ref "docs/deployment/config" >}}#execution-batch-job-recovery-previous-worker-recovery-timeout):
+  This setting determines the timeout duration allowed for Shuffle workers to reconnect. During the recovery process, Flink 
+  requests the retained intermediate result data information from the Shuffle Master. If the timeout is reached, 
+  Flink will use all the acquired intermediate result data to recover the state.
+- [job-event.store.write-buffer.flush-interval]({{< ref "docs/deployment/config" >}}#job-event-store-write-buffer-flush-interval):
+  This setting determines the flush interval for the JobEventStore's write buffers.
+- [job-event.store.write-buffer.size]({{< ref "docs/deployment/config" >}}#job-event-store-write-buffer-size): This 
+  setting determines the write buffer size in the JobEventStore. When the buffer is full, its contents are flushed to the external
+  filesystem.
+
+### Enable batch jobs progress recovery for sources
+
+Currently, only the new source (FLIP-27) supports progress recovery for batch jobs. To enable this feature, the SplitEnumerator of a new source (FLIP-27) need implement the SupportsBatchSnapshot interface:
+
+````
+public interface SupportsBatchSnapshot {}
+````
+
+This interface indicates that the SplitEnumerator supports taking snapshots in batch processing scenarios. 
+If a source's SplitEnumerator does not implement this interface and the job master goes through a failover, one of the following two situations will occur:
+1. If not all tasks of this source are finished, we will reset and re-run all these tasks.
+2. If all tasks of this source are finished, no additional action is required, and the job can continue to run.
+However, if any of these tasks need to be restarted at some point in the future (for example, due to a 
+PartitionNotFound exception), then all subtasks of this source will need to be reset and rerun.
+
+## Limitations
+
+- Only working with the new source (FLIP-27): Since the legacy source has been deprecated, this feature is only
+  working with the new source.
```

Comment:
this feature is only working with the new source.
-> this feature only supports the new source.